src/map/latent_effect_container.cpp

```diff
@@ -1040,52 +1047,59 @@ bool CLatentEffectContainer::ProcessLatentEffect(CLatentEffect& latentEffect)
         case LATENT::NATION_CONTROL:
         {
             // player is logging in/zoning
-            if (m_POwner->loc.zone == nullptr)
+            // checking the state of the user's current zone and
+            // the destination zone in tandem seems to work.
+            if (m_POwner->loc.zone == nullptr && static_cast<uint16_t>(m_POwner->loc.destination) == 0)
             {
                 break;
             }
 
-            auto region      = m_POwner->loc.zone->GetRegionID();
+            // Grab the user's destination if they're zoning.
+            // Otherwise, grab their current zone.
+            auto region      = m_POwner->loc.zone == nullptr ? zoneutils::GetCurrentRegion(m_POwner->loc.destination) : zoneutils::GetCurrentRegion(playerZoneID);
             auto hasSignet   = m_POwner->StatusEffectContainer->HasStatusEffect(EFFECT_SIGNET);
             auto hasSanction = m_POwner->StatusEffectContainer->HasStatusEffect(EFFECT_SANCTION);
             auto hasSigil    = m_POwner->StatusEffectContainer->HasStatusEffect(EFFECT_SIGIL);
-
+            auto regionAlwaysOutOfControl = zoneutils::IsAlwaysOutOfNationControl(region);
```

Comment:
Formatting

Reply:
Fixed it. Odd my linter did not auto-fix this.